src/transformers/training_args.py

```diff
@@ -2049,32 +2055,62 @@ def _setup_devices(self) -> "torch.device":
                     f"Using the `Trainer` with `PyTorch` requires `accelerate>={ACCELERATE_MIN_VERSION}`: "
                     "Please run `pip install transformers[torch]` or `pip install accelerate -U`"
                 )
+        # We delay the init of `PartialState` to the end for clarity
+        accelerator_state_kwargs = {"enabled": True, "use_configured_state": False}
+        if isinstance(self.accelerator_config, AcceleratorConfig):
+            accelerator_state_kwargs["use_configured_state"] = self.accelerator_config.pop(
+                "use_configured_state", False
+            )
+        if accelerator_state_kwargs["use_configured_state"]:
+            if PartialState._shared_state == {}:
+                raise ValueError(
+                    "Passing `'use_configured_state':True` to the AcceleratorConfig requires a pre-configured "
+                    "`AcceleratorState` or `PartialState` to be defined before calling `TrainingArguments`. "
+                )
+            # We rely on `PartialState` to yell if there's issues here (which it will)
+            self.distributed_state = PartialState(cpu=self.use_cpu)
```

Comment:
this still doesn't account for the case when user passed `--fsdp` but hasn't enabled it via `PartialState`. In general, my comment was about the mismatch between the training arguments vs the PartialState set by them.

Reply:
If we just have a `PartialState` you can still initialize FSDP later as it's done at the `AcceleratorState` level. I'll do a quick test to verify, but the `PartialState` doesn't care about FSDP

Reply:
(PartialState only initializes the distributed env, for things like FSDP or DeepSpeed plugin that happens later, though DS will still be called/activated if the env is set properly with the PartialState)

Reply:
See here: https://github.com/huggingface/accelerate/blob/main/src/accelerate/state.py#L894-L904

Reply:
Understood, Thank you Zach for the clarifications.